File: app/src/main/java/com/callmangement/ui/errors/adapter/RemarkAdapter.java
package com.callmangement.ui.errors.adapter;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.callmangement.databinding.ItemRemarkBinding;
import com.callmangement.ui.errors.model.GetRemarkDatum;

import java.util.ArrayList;

public class RemarkAdapter extends RecyclerView.Adapter<RemarkAdapter.ViewHolder> {
    private final Context context;
    private final ArrayList<GetRemarkDatum> getRemarkDatumArrayList;

    public RemarkAdapter(Context context, ArrayList<GetRemarkDatum> getRemarkDatumArrayList) {
        this.context = context;
        this.getRemarkDatumArrayList = getRemarkDatumArrayList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemRemarkBinding binding = ItemRemarkBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false);
        return new ViewHolder(binding);
    }
    public void makeToast(String string) {
        if (TextUtils.isEmpty(string)) return;
        Toast.makeText(context.getApplicationContext(), string, Toast.LENGTH_SHORT).show();

    }


   /* public void showProgress(String message) {
        hideProgress();
        mDialog = new ProgressDialog(context.getApplicationContext());
        mDialog.setMessage(message);
        mDialog.setCancelable(false);
        mDialog.show();
    }

    public void showProgress() {
        hideProgress();
        mDialog = new ProgressDialog(context.getApplicationContext());
        mDialog.setMessage("Loading...");
        mDialog.setCancelable(false);
        mDialog.show();
    }

    public void hideProgress() {
        try {
            if (mDialog!=null) mDialog.dismiss();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }*/

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        GetRemarkDatum model = getRemarkDatumArrayList.get(position);
        holder.binding.remark.setText(model.getRemark());
  //    holder.binding.createdby.setText(model.getCreatedBy().toString());
        String createdon = String.valueOf(model.getRemarkDate());
        if (createdon!=null)
        {
            holder.binding.createdon.setText(createdon);
            // holder.binding.createdby.setText(createdby);
        }
        else
        {

        }

      String createdby = String.valueOf(model.getUserType());
        if (createdby!=null)
        {
            holder.binding.createdby.setText(createdby);
            // holder.binding.createdby.setText(createdby);
        }
        else
        {

        }
        holder.binding.status.setText(model.getErrorStatus());


       /* else if (model.getErrorStatusId() == 3) {
            holder.binding.errortype.setText(model.getErrorStatus());
            holder.binding.errortype.setTextColor(context.getResources().getColor(R.color.colorGreenDark));
        }*/



    /*    holder.binding.rlItem.setOnClickListener(view -> {
            Intent intent = new Intent(context, ErrorposDetail.class);
            intent.putExtra("param",  model);
            context.startActivity(intent);
        });*/


      //  holder.binding.t

    }

    @Override
    public int getItemCount() {
        return getRemarkDatumArrayList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemRemarkBinding binding;
        public ViewHolder(ItemRemarkBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

}

File: app/src/main/java/com/callmangement/ui/errors/adapter/RemarkAdapter.kt
package com.callmangement.ui.errors.adapter

import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.callmangement.databinding.ItemRemarkBinding
import com.callmangement.ui.errors.model.GetRemarkDatum

class RemarkAdapter(
    private val context: Context,
    private val getRemarkDatumArrayList: ArrayList<GetRemarkDatum>
) :
    RecyclerView.Adapter<RemarkAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemRemarkBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    fun makeToast(string: String?) {
        if (TextUtils.isEmpty(string)) return
        Toast.makeText(context.applicationContext, string, Toast.LENGTH_SHORT).show()
    }


    /* public void showProgress(String message) {
        hideProgress();
        mDialog = new ProgressDialog(context.getApplicationContext());
        mDialog.setMessage(message);
        mDialog.setCancelable(false);
        mDialog.show();
    }

    public void showProgress() {
        hideProgress();
        mDialog = new ProgressDialog(context.getApplicationContext());
        mDialog.setMessage("Loading...");
        mDialog.setCancelable(false);
        mDialog.show();
    }

    public void hideProgress() {
        try {
            if (mDialog!=null) mDialog.dismiss();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }*/
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = getRemarkDatumArrayList[position]
        holder.binding.remark.text = model.remark
        //    holder.binding.createdby.setText(model.getCreatedBy().toString());
        val createdon = model.remarkDate.toString()
        if (createdon != null) {
            holder.binding.createdon.text = createdon
            // holder.binding.createdby.setText(createdby);
        } else {
        }

        val createdby = model.userType.toString()
        if (createdby != null) {
            holder.binding.createdby.text = createdby
            // holder.binding.createdby.setText(createdby);
        } else {
        }
        holder.binding.status.text = model.errorStatus


        /* else if (model.getErrorStatusId() == 3) {
            holder.binding.errortype.setText(model.getErrorStatus());
            holder.binding.errortype.setTextColor(context.getResources().getColor(R.color.colorGreenDark));
        }*/


        /*    holder.binding.rlItem.setOnClickListener(view -> {
            Intent intent = new Intent(context, ErrorposDetail.class);
            intent.putExtra("param",  model);
            context.startActivity(intent);
        });*/


        //  holder.binding.t
    }

    override fun getItemCount(): Int {
        return getRemarkDatumArrayList.size
    }

    class ViewHolder(val binding: ItemRemarkBinding) : RecyclerView.ViewHolder(
        binding.root
    )
}
